src/shims/native_lib/trace/parent.rs

```diff
@@ -417,71 +412,13 @@ fn handle_segfault(
                             if acc_ty.is_writable() {
```

Comment:
Can it actually happen that the same operation is both a read and a write? Is that used for instance for atomic fetch-and-add and operations like that?

Reply:
Well, that's only on the RCW events. I can add a boolean `is_certain` field on the write and if it's false we double-check that the allocation it's writing to is mutable first, but that will be a very rare edge case

Reply:
I'll pour over the iced source later then, just to see which instructions that flag is set for. Ty!